app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -229,6 +231,10 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
         components.publicSuffixList.prefetch()
 
+        // Changing a language on the Language screen restarts the activity, but the activity keeps
+        // the old layout direction. We have to update the direction manually.
+        window.decorView.layoutDirection = TextUtils.getLayoutDirectionFromLocale(Locale.getDefault())
```

Comment:
Drive by: This seems like a last resort location to do an initialization step from a setting. Would it be possible to include this in the locale settings flow when updating the language? `DefaultLocaleSettingsController` might be a better alternative.

Reply:
That was my first though as well, but this approach has an issue – invalidating the layout and recreating the activity happens not synchronously and looks like a glitch.

https://user-images.githubusercontent.com/92760693/211407358-b70ea490-f7b1-4cff-86e7-bfb4c22c5382.mp4

What concerned did you have in mind here? Applying proper orientation during `onCreate` looks okay to me. `setLayoutDirection` has also a safety check to invalidate the layout only if it is needed.

Reply:
Where is the orientation being set in HomeActivity?

My primary concern is that this was even needed at all and might be a heavy hammer, but I did notice that `HomeActivity` extends from `LocaleAwareAppCompatActivity` which has this same line in its own `onCreate`. Based on the adjacent comment in `HomeActivity`, the `super.onCreate` only gets called after some disk read violations, but it was obviously needed.

Reply:
The use-case I am working on is `DefaultLocaleSettingsController` changing the Locale to something that has a different text orientation and restarts the activity. Changing the locale should also change the layout orientation, but it doesn't and the activity is recreated with the old orientation. To avoid glitching, enforcing changed orientation looks like a good idea to me. What do you think?

Reply:
Judging by the conversation in https://github.com/mozilla-mobile/fenix/issues/9413, I think I'm OK with this approach, as it's a natural side effect of us having to take care of all of the configuration changes. Maybe it won't be necessary in a 100% Compose world 🤞